test/stripe/list_object_test.rb

```diff
@@ -25,20 +25,80 @@ class ListObjectTest < Test::Unit::TestCase
       assert_equal expected, list.each.to_a
     end
 
-    should "provide #auto_paging_each" do
+    should "provide #reverse_each" do
       arr = [
         { id: 1 },
         { id: 2 },
         { id: 3 },
       ]
+      expected = Util.convert_to_stripe_object(arr.reverse, {})
+      list = Stripe::ListObject.construct_from(data: arr)
+      assert_equal expected, list.reverse_each.to_a
+    end
+
+    should "provide #auto_paging_each that supports forward pagination" do
+      arr = [
+        { id: 1 },
+        { id: 2 },
+        { id: 3 },
+        { id: 4 },
+        { id: 5 },
+        { id: 6 },
+      ]
       expected = Util.convert_to_stripe_object(arr, {})
 
+      # Initial list object to page on. Notably, its last data element will be
+      # used as a cursor to fetch the next page.
       list = TestListObject.construct_from(data: [{ id: 1 }],
                                            has_more: true,
                                            url: "/things")
+      list.filters = { limit: 3 }
+
+      # The test will start with the synthetic list object above, and use it as
+      # a starting point to fetch two more pages. The second page indicates
+      # that there are no more elements by setting `has_more` to `false`, and
+      # iteration stops.
       stub_request(:get, "#{Stripe.api_base}/things")
-        .with(query: { starting_after: "1" })
-        .to_return(body: JSON.generate(data: [{ id: 2 }, { id: 3 }], has_more: false))
+        .with(query: { starting_after: "1", limit: "3" })
+        .to_return(body: JSON.generate(data: [{ id: 2 }, { id: 3 }, { id: 4 }], has_more: true, url: "/things"))
+      stub_request(:get, "#{Stripe.api_base}/things")
+        .with(query: { starting_after: "4", limit: "3" })
+        .to_return(body: JSON.generate(data: [{ id: 5 }, { id: 6 }], has_more: false, url: "/things"))
+
+      assert_equal expected, list.auto_paging_each.to_a
+    end
+
+    should "provide #auto_paging_each that supports backward pagination with `ending_before`" do
+      arr = [
+        { id: 6 },
+        { id: 5 },
+        { id: 4 },
+        { id: 3 },
+        { id: 2 },
+        { id: 1 },
+      ]
```

Comment:
And this is the new "backward" iteration test.

One thing to note about both tests is that the `id` series I'm using here isn't an _amazing_ analogy for how the API actually works because in practice lists in Stripe API are inherently inverted in their natural state. So e.g., if I create a new charge and it comes out with `id: 7`, it's actually the first item in `GET /v1/charges` rather than the last.

So it might be less confusing if the forward pagination used this series (6..1) and backward pagination used 1..6 because backwards pagination is iterating backwards through a backwards list.

That said, I think it'd only less confusing if you're already very aware of how the API's list semantics work already, and for pretty much anyone else, seeing 6..1 in a test for "backwards" is more intuitive, which is why I did it this way.